src/assisted_test_infra/test_infra/helper_classes/cluster.py

```diff
@@ -1292,3 +1293,79 @@ def wait_and_kill_installer(self, host):
+        host_network = {}
+        for host in hosts_list:
+            log.debug(f"Validation host {host['requested_hostname']}")
+            inventory = yaml.safe_load(host["inventory"])
+
+            interfaces = inventory["interfaces"]
+            for interface in interfaces:
+                for _, address_version in consts.IP_VERSIONS.items():
+                    if address_version in interface.keys():
+                        if len(interface[address_version]) != 0:
+                            host_network[interface["mac_address"]] = {address_version: interface[address_version]}
+        log.debug(f"host_network {host_network}")
+        return host_network
+
+    def validate_static_ip(self):
+        if self._infra_env_config.static_network_config is None:
+            log.debug("Skipping static IP validation")
+            return
+
+        log.info("Starting static IP validation")
+        self.wait_until_hosts_are_discovered()
+
+        host_network = self.format_host_current_network()
+        if host_network == {}:
+            raise Exception("Couldn't find host network configurations")
+
+        host_failure = []
+        for expected_host_mapping in yaml.safe_load(str(self._infra_env_config.static_network_config)):
+            for expected_host_interfaces in expected_host_mapping["mac_interface_map"]:
+                expected_interface_name, expected_interface_mac = (
+                    expected_host_interfaces["logical_nic_name"],
+                    expected_host_interfaces["mac_address"],
+                )
+            log.debug(f"Validating expected interface {expected_interface_name} with mac {expected_interface_mac}")
+            if expected_interface_mac not in host_network.keys():
+                host_failure.append(f"missing mac address {expected_interface_mac}")
+                continue
+
+            network_yaml = yaml.safe_load(return_yaml_from_string(expected_host_mapping["network_yaml"]))
+            interfaces = network_yaml["interfaces"]
+            expected_hosts_address = []
+            for current_interface in interfaces:
+                if current_interface["name"] == expected_interface_name:
+                    for address_version in consts.IP_VERSIONS.keys():
+                        if address_version in current_interface.keys():
+                            if len(current_interface[address_version]["address"]) == 0:
+                                continue
+                            addresses = [
+                                f'{item["ip"]}/{item["prefix-length"]}'
+                                for item in current_interface[address_version]["address"]
+                            ]
+                            expected_hosts_address.append({expected_interface_mac: addresses})
+                            for addr in addresses:
+                                if addr in host_network[expected_interface_mac][consts.IP_VERSIONS[address_version]]:
```

Comment:
I think you need to check that the keys `expected_interface_mac` and `consts.IP_VERSIONS[address_version]` exist in their map, otherwise python will raise an exception, and it may complicate the error message in jobs.

Reply:
so I've checked on the config both version are there and empty be default - I had that check and I removed it 

we can added it back